src/main/drivers/pwm_mapping.c

```diff
@@ -643,133 +645,145 @@ pwmIOConfiguration_t *pwmGetOutputConfiguration(void){
-            continue;
-        }
+    if (init->useCurrentMeterADC && timerHardwarePtr->gpio == CURRENT_METER_ADC_GPIO && timerHardwarePtr->pin == CURRENT_METER_ADC_GPIO_PIN) {
+        return true;
+    }
 #endif
 
 #ifdef SONAR
-        if (init->sonarGPIOConfig && timerHardwarePtr->gpio == init->sonarGPIOConfig->gpio &&
-            (
-                timerHardwarePtr->pin == init->sonarGPIOConfig->triggerPin ||
-                timerHardwarePtr->pin == init->sonarGPIOConfig->echoPin
-            )
-        ) {
-            continue;
-        }
+    if (init->sonarGPIOConfig && timerHardwarePtr->gpio == init->sonarGPIOConfig->gpio &&
+        (
+            timerHardwarePtr->pin == init->sonarGPIOConfig->triggerPin ||
+            timerHardwarePtr->pin == init->sonarGPIOConfig->echoPin
+        )
+    ) {
+        return true;
+    }
 #endif
 
-        // hacks to allow current functionality
+    // hacks to allow current functionality
 
 #if defined(SPRACINGF3MINI)
-            // remap PWM1, 6-9 as PWM input when parallel PWM is used (for AUX1 and RC1-4, respectively)
-            if (init->useParallelPWM && (timerIndex == PWM6 || timerIndex == PWM7 || timerIndex == PWM8 || timerIndex == PWM9 || timerIndex == PWM1)) {
-                type = MAP_TO_PWM_INPUT;
-            }
+    // remap PWM1, 6-9 as PWM input when parallel PWM is used (for AUX1 and RC1-4, respectively)
+    if (init->useParallelPWM && (timerIndex == PWM6 || timerIndex == PWM7 || timerIndex == PWM8 || timerIndex == PWM9 || timerIndex == PWM1)) {
+        type = MAP_TO_PWM_INPUT;
+    }
+#else
+    UNUSED(timerIndex);
 #endif
 
-        if (type == MAP_TO_PWM_INPUT && !init->useParallelPWM)
-            continue;
+    if (type == MAP_TO_PWM_INPUT && !init->useParallelPWM) {
+        return true;
+    }
+
+    if (type == MAP_TO_PPM_INPUT && !init->usePPM) {
+        return true;
+    }
+
+    return false;
+}
+
+pwmIOConfiguration_t *pwmInit(const drv_pwm_config_t *init)
+{
+    int i = 0;
+    const uint16_t *setup;
+
+    int channelIndex = 0;
+
+
+    memset(&pwmIOConfiguration, 0, sizeof(pwmIOConfiguration));
+
+    // this is pretty hacky shit, but it will do for now. array of 4 config maps, [ multiPWM multiPPM airPWM airPPM ]  PWM mappings are used for RX_MSP.
+    if (init->airplane)
+        i = 2; // switch to air hardware config
+    if (init->usePPM || init->useSerialRx)
+        i++; // next index is for PPM
+
+    setup = hardwareMaps[i];
 
-        if (type == MAP_TO_PPM_INPUT && !init->usePPM)
+    for (i = 0; i < USABLE_TIMER_CHANNEL_COUNT && setup[i] != 0xFFFF; i++) {
+        const uint8_t timerIndex = setup[i] & 0x00FF;
+        uint8_t type = (setup[i] & 0xFF00) >> 8;
+        const timerHardware_t *timerHardwarePtr = &timerHardware[timerIndex];
+
+        if (pwmIsUsed(init, type, timerHardwarePtr, timerIndex)) {
```

Comment:
Not sure that `pwmIsUsed` is a good name, it doesn't tell that it's actually a check if we can use a given pwm output/input. Maybe a better name is `pwmChannelCanBeUsed` ?